src/megaclient.cpp

```diff
@@ -12364,7 +12360,8 @@ bool MegaClient::syncdown(LocalNode* l, string* localpath, bool rubbish)
 
                     rit->second->syncget = new SyncFileGet(l->sync, rit->second, localpath);
                     nextreqtag();
-                    startxfer(GET, rit->second->syncget);
+                    DBTableTransactionCommitter committer(tctable); // TODO: use one committer for all files in the loop, without calling syncdown() recursively
```

Comment:
Missing beginOnce? a `filecacheadd` that leads to `tctable->put` might be executed within starxfer.  In any case, I'm missing where the tctable->begin was called in the past in this case. I wonder if this requires a committer at all.

Reply:
excellent point, and I'm working on extending the changes to cover those cases.  thanks!